### ISSUE TITLE
CD script not taking into account integration tests

### ISSUE THREAD

**Issue by edublancas** [ploomber/jupysql#514]
github actions is configured to automatically upload to PyPI whenever we tag a commit with a version:![image](https://github.com/ploomber/jupysql/assets/989250/95324b35-4f9a-4476-b7b2-82326257cca9)as you can see [tests must pass](https://github.com/ploomber/jupysql/actions/runs/5074031711) before we release something. however, we are currently ignoring integration tests since they're defined in a different file and I didn't find a way to make the release job dependent on those as well.we should fix this and ensure that we only release when _all tests_ pass. I don't know if there are better solutions than putting all our tests in a single file

**Comment by tonykploomber** [comment]
I raised [the PR](https://github.com/ploomber/jupysql/pull/520) to see if workflow-call can be used when we have this dependency workflow requirement

Ref: https://stackoverflow.com/questions/58457140/dependencies-between-workflows-on-github-actions

**Comment by edublancas** [comment]
I think we'd need to merge this to test it, right? because the release workflow won't be triggered from the PR

**Comment by edublancas** [comment]
can you check the CI? @tonykploomber 

I'm not sure if this is working, see this https://github.com/ploomber/jupysql/actions/runs/5094281546/jobs/9157899697 - they scripts take 0s (is this expected?)

**Comment by tonykploomber** [comment]
I think it's looks good from my side?

This is the all [integration-test-live](https://github.com/ploomber/jupysql/actions/workflows/ci-integration-db-live.yaml) build results, and latest build [Syntactic sugar for outputting result variables](https://github.com/ploomber/jupysql/actions/runs/5094978371/jobs/9159430089) (#501) looks it does execute the integration testing

<img width="1859" alt="Screenshot 2023-05-26 at 9 18 17 PM" src="https://github.com/ploomber/jupysql/assets/123580782/d4fcff58-436d-4707-b0b9-db612a26a9a4">

**Comment by edublancas** [comment]
Ok, I'm confused about how the CI jobs work. These three jobs were all triggered due to a merge in master:

CI - DB integration - Cloud looks ok, testing takes ~4m: https://github.com/ploomber/jupysql/actions/runs/5099986696/jobs/9168000128

What is CI - DB Integration - Local for? Integration test only takes 1s. It's only printing the sqlalchemy version: https://github.com/ploomber/jupysql/actions/runs/5099986694/jobs/9168000126

For the unit tests: https://github.com/ploomber/jupysql/actions/runs/5099986708

integration-test-live takes 5s to run from start to finish
and integration-test-non-live takes 3m to install stuff but then integration testing takes 0s

**Comment by edublancas** [comment]
I merged #535:

cloud and local are now working:

**Comment by edublancas** [comment]
I merged #535:

cloud and local are now working - what's the difference between these two?:
https://github.com/ploomber/jupysql/actions/runs/5113853634
https://github.com/ploomber/jupysql/actions/runs/5113853633

non-live in the unit tests also doing stuff:
https://github.com/ploomber/jupysql/actions/runs/5113853641/jobs/9193515718

live still showing 0:
https://github.com/ploomber/jupysql/actions/runs/5113853641/jobs/9193515656

is this expected?

**Comment by tonykploomber** [comment]
I just realize after #520 got merged, this will generate 4 CI actions
- CI - DB Integration - Cloud
- CI - DB Integration - Local
- CI / integration-test-live (generated by `.github/workflows/ci.yaml`)
- CI / integration-test-non-live (generated by `.github/workflows/ci.yaml`)

However, those are duplicated.

I will see if we can use [workflow_run](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#workflow_run) to handle dependency issue instead of creating new duplicated jobs

**Comment by tonykploomber** [comment]
New PR raised in https://github.com/ploomber/jupysql/pull/549

I tried three ways to solve the dependency issue, but none of them seems to work on our project:
- [workflow_run](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#workflow_run)
- [Wait On Check](https://github.com/lewagon/wait-on-check-action)
- [Wait For Check](https://github.com/marketplace/actions/wait-for-check)

Eventually, we can actually just use `ci.yaml` to trigger two integration testing (so this will guarantee the release job depends on those executed results). Also we need to remove the `push` and `pull_request` in original workflow, so it wouldn't have any duplicated workflows

**Comment by edublancas** [comment]
@tonykploomber: if I understand correctly, the solution in #549 keeps the integration in separate files but ci.yml triggers them? it's essentially keeping thins separate and organized but having a central place to trigger events?

> CI - DB Integration - Cloud
> CI - DB Integration - Local

what's the difference between these two workflows?

**Comment by tonykploomber** [comment]
@edublancas 
1. Yes, you are correct. The reason why we see it's duplicating is there is on push or pull-request event to trigger the workflow. Now we only trigger by the calling from ci.yaml
2.
`CI - DB Integration - Cloud` is majorly executing the snowflake part, we need to use sqlalchemy 1.4 to support that
`CI - DB Integration - Local ` is testing the databases by containers, sqlalchemy 2 was used 

Major reason to split those workflows is about the versioning of sqlalchemy
